frontend/src/metabase/query_builder/components/NativeQueryEditor/DataSourceSelectors/DataSourceSelectors.jsx

```diff
@@ -135,21 +149,14 @@ const checkIfThereAreMultipleDatabases = (database, databases) =>
   database == null ||
   (databases.length > 1 && databases.some(db => db.id === database.id));
```

Comment:
we might need to split this check into 2 checks, and have a case for when the selected db isn't int he list of dbs.